accumulator.go

```diff
@@ -144,6 +147,18 @@ func (p *Pollard) calculateNewRoot(node *polNode) *polNode {
 		// Roots point to their children. Those children become nieces here.
 		swapNieces(root, node)
 
+		// Check if the current node should be remembered
+		if node.remember {
+			// If the sibling needs to be remembered, store this node
+			sibling, err := node.getSibling()
+			if err == nil && sibling != nil && sibling.remember {
+				p.RequiredNodes = append(p.RequiredNodes, node)
+			}
+
+			// Recursively check for aunt nodes
+			p.checkAuntNodes(node)
```

Comment:
No need to check for aunt nodes during additions since the aunts of this newly added node will be the current roots. We'll always have roots.